opencog/spacetime/octomap/TimeOctomap.h

```diff
@@ -81,23 +82,38 @@ struct TimeSlice
         map_tree.setNodeData(location, ato);
     }
 
+    void insert_atom(const T& ato){
+        temporal.push_back(ato);
+    }
+
     // Remove the atom from this time-slice.
-    void remove_atom(const T& ato){
-        point3d_list pl;
-        for (typename AtomOcTree<T>::tree_iterator it2 = map_tree.begin_tree(),
-                endit2 = map_tree.end_tree();
-                it2 != endit2;
-                ++it2)
-        {
-            if (it2->getData() == ato)
-            {
-                pl.push_back(it2.getCoordinate());
-                it2->setData(T()); //FIXME this requires default constructor always??
+    void remove_atom(const T& ato, bool has_xyz=true){
```

Comment:
I really really don't like the `has_xyz` on remove.  You were already told whether or not the atom had a location, when you inserted it. Why do you need to be told s second time?  This forces the user to remember something, so that they can tell you something you already know.

For example, it makes the attention-allocation forgetting agent more complicated, because the forgetting agent would have to somehow figure out if the thing to be forgotten has an xyz position. That's just .. crazy.